src/main.rs

```diff
@@ -141,25 +132,36 @@ async fn main() {
         },
         on_error: |error| Box::pin(on_error(error)),
         skip_checks_for_owners: false,
-        event_handler: |ctx, event, framework, data| {
-            Box::pin(event_handler(ctx, event, framework, data))
-        },
+        event_handler: |ctx, event, framework, data| Box::pin(event_handler(ctx, event, framework, data)),
         ..Default::default()
     };
 
     let framework = poise::Framework::builder()
         .setup(move |ctx, _ready, framework| {
             Box::pin(async move {
-                poise::builtins::register_globally(ctx, &framework.options().commands).await?;
-                Ok(Data {})
+                poise::builtins::register_globally(ctx, &framework.options().commands)
+                    .await
+                    .map_err(|e| e.into())
+                    .map(|_| Data {})
```

Comment:
You can define this as `struct Data;` above and then just refer to it without `{}`.